testing/src/main/java/org/oppia/android/testing/logging/EventLogSubject.kt

```diff
@@ -1944,6 +1965,126 @@ class EventLogSubject private constructor(
     }
   }
 
+  /**
+   * Truth subject for verifying properties of [EventLog.FeatureFlagContext]s.
+   *
+   * Note that this class is also a [LiteProtoSubject] so other aspects of the underlying
+   * [EventLog.FeatureFlagContext] proto can be verified through inherited methods.
+   *
+   * Call [FeatureFlagContextSubject.assertThat] to create the subject.
+   */
+  class FeatureFlagContextSubject private constructor(
+    metadata: FailureMetadata,
+    private val actual: EventLog.FeatureFlagContext
+  ) : LiteProtoSubject(metadata, actual) {
+    /**
+     * Returns a [StringSubject] to test [EventLog.FeatureFlagContext.uniqueUserUuid].
+     *
+     * This method never fails since the underlying property defaults to empty string if it's not
+     * defined in the context.
+     */
+    fun hasUniqueUserUuidThat(): StringSubject = assertThat(actual.uniqueUserUuid)
+
+    /**
+     * Returns a [StringSubject] to test [EventLog.FeatureFlagContext.sessionId].
+     *
+     * This method never fails since the underlying property defaults to empty string if it's not
+     * defined in the context.
+     */
+    fun hasSessionIdThat(): StringSubject = assertThat(actual.sessionId)
+
+    /**
+     * Returns a [IterableSubject] to test [EventLog.FeatureFlagContext.getFeatureFlagsList].
+     *
+     * This method never fails since the underlying property defaults to empty list if it's not
+     * defined in the context.
+     */
+    fun hasFeatureFlagListThat(): IterableSubject = assertThat(actual.featureFlagsList)
+
+    /**
+     * Returns a [FeatureFlagItemContextSubject] to test
+     * [EventLog.FeatureFlagContext.getFeatureFlagsList].
+     *
+     * This method never fails since the underlying property defaults to empty string if it's not
+     * defined in the context.
+     */
+    fun hasFeatureFlagItemContextThatHasName(flagName: String): FeatureFlagItemContextSubject =
+      FeatureFlagItemContextSubject.assertThat(
+        actual.featureFlagsList.firstOrNull { it.flagName == flagName }
```

Comment:
@BenHenning, I had the thought that we might be able to create a function that can return a single `FeatureFlagItemContext` from the list
(`hasFeatureFlagListThat()`), and we can subsequently chain the assertions e.g.:
  
```kotlin
assertThat(eventLog).hasFeatureFlagContextThat { 
hasFeatureFlagListThat(){
hasFeatureFlagItemContextThat(){
hasFeatureFlagNameThat().isEqualTo(TEST_FEATURE_FLAG_WITH_ENABLED_DEFAULTS)
hasFeatureFlagEnabledStateThat().isEqualTo(true)    hasFeatureFlagSyncStateThat().isEqualTo(SyncStatus.SYNCED_FROM_SERVER)
          }
      }
}
```

I think that the bit that @kkmurerwa has a challenge with is retrieving a single item from the iterable returned by `hasFeatureFlagListThat()`. Do you have pointers to how we can achieve this?